bench/tx-generator/src/Cardano/Benchmarking/Script/Core.hs

```diff
@@ -632,6 +643,37 @@ createChangeGeneric submitMode createCoins addressMsg value count = do
   chunkList _ [] = []
   chunkList n xs = as : chunkList n bs where (as,bs) = splitAt n xs
 
+spendAutoScript :: SubmitMode -> FilePath -> ThreadName -> NumberOfTxs -> TPSRate -> ActionM ()
+spendAutoScript submitMode scriptFile threadName txCount tps = do
+  protocolParameters <- queryProtocolParameters
+  budget <- case protocolParamMaxTxExUnits protocolParameters of
+    Nothing -> throwE $ ApiError "cannot determine protocolParamMaxTxExUnits"
+    Just b -> return b
+  script <- liftIO $ readScript scriptFile
+  let
+    costFn :: Integer -> Either String Bool
+    costFn n = case preExecuteScript protocolParameters script (ScriptDataNumber 0) (ScriptDataNumber $ n + 1000000) of
+      Left err -> Left err
+      Right use -> Right $ (executionSteps use <= executionSteps budget) && (executionMemory use <= executionMemory budget)
+
+  redeemer <- case startSearch costFn 0 100000 of
+    Left err -> throwE $ ApiError $ "cannot find fitting redeemer :" ++ err
+    Right x -> return $ ScriptDataNumber $ x + 1000000
+  runPlutusBenchmark submitMode scriptFile PreExecuteScript (ScriptDataNumber 0) redeemer threadName txCount tps
+  where
+    startSearch f a b = do
+       l <- f a
+       h <- f b
+       if not l || h
+          then Left $ "binary search: bad inital bounds : " ++ show (a,b)
```

Comment:
I think we might benefit from a more clear error message here -- since there are two distinct cases:

1. the sought value is above the upper boundary
2. the sought value is below the lower boundary

Reply:
Right the sought value is below the lower boundary,
or the thought value is above the upper boundary....